sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp

```diff
@@ -85,13 +85,18 @@ extern "C" constexpr __spv::MatrixLayout joint_matrix_layout_to_spv(
   }
 }
 
-template<typename Ta, typename Tb, typename Tc>
+template <typename Ta, typename Tb, typename Tc, typename Td>
 constexpr uint32_t CalculateMatrixOperand() {
   if constexpr (std::is_same<Ta, sycl::ext::oneapi::bfloat16>::value &&
-                std::is_same<Tb, sycl::ext::oneapi::bfloat16>::value &&
-                std::is_same<Tc, float>::value)
+                std::is_same<Tb, sycl::ext::oneapi::bfloat16>::value)
     return static_cast<uint32_t>(
         __spv::MatrixOperands::MatrixAAndBBFloat16ComponentsINTEL);
+  if constexpr (std::is_same<Tc, sycl::ext::oneapi::bfloat16>::value)
```

Comment:
what if A, B and C are all bf16? should we return `MatrixAAndBBFloat16ComponentsINTEL + MatrixCBFloat16ComponentsINTEL` in this case?
and similar question for other combinations:
A and B and D - bfloat16
A and B and C and D - bfloat16
seems like different resulting operand value should be used.